CompetitionTemplate/tests/competition/subsystems/drive/MonitorDefenseTraversalTest.java

```diff
@@ -0,0 +1,52 @@
+package competition.subsystems.drive;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+import competition.subsystems.drive.commands.MonitorDefenseTraversalModule;
+import competition.subsystems.drive.commands.MonitorDefenseTraversalModule.DefenseState;
+import edu.wpi.first.wpilibj.MockTimer;
+
+public class MonitorDefenseTraversalTest extends DriveTestBase {
+        
+    MonitorDefenseTraversalModule module;
+    MockTimer timer;
+    
+    @Before
+    public void setup() {
+        super.setup();
+        timer = injector.getInstance(MockTimer.class);
+        module = injector.getInstance(MonitorDefenseTraversalModule.class);
+    }
+    
+    @Test
+    public void testStartNotOnDefense() {
+        assertTrue(module.measureState(0) == DefenseState.NotOnDefense);
+    }
+    
+    @Test
+    public void testSuddenTilt() {
+        assertTrue(module.measureState(50) == DefenseState.OnDefense);
+    }
+    
+    @Test
+    public void testTiltFollowedByFlat() {
+        assertTrue(module.measureState(-50) == DefenseState.OnDefense);
```

Comment:
It's scary that "measyreState" stores state about the value you give it. Can we change it to "updateState" or similar?